go/interpreter/lfvm/instructions_test.go

```diff
@@ -1191,6 +1191,87 @@ func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
 	}
 }
 
+func TestGenericCreate_ReportsErors(t *testing.T) {
+
+	u64Overflow := new(uint256.Int).Add(uint256.NewInt(math.MaxUint64), uint256.NewInt(math.MaxUint64))
+	one := uint256.NewInt(1)
+
+	tests := map[string]struct {
+		offset, size  uint256.Int
+		kind          tosca.CallKind
+		revision      tosca.Revision
+		expectedError error
+	}{
+		"offset overflow": {
+			offset:        *u64Overflow,
+			size:          *one,
+			expectedError: errOverflow,
+		},
+		"size overflow": {
+			offset:        *one,
+			size:          *u64Overflow,
+			expectedError: errOverflow,
+		},
+		"not enough gas for code size": {
+			offset:        *one,
+			size:          *uint256.NewInt(31),
+			revision:      tosca.R12_Shanghai,
+			expectedError: errOutOfGas,
+		},
+		"not enough gas for create 2 init code hashing": {
+			offset:        *one,
+			size:          *one,
+			kind:          tosca.Create2,
+			revision:      tosca.R11_Paris,
+			expectedError: errOutOfGas,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := context{
+				params: tosca.Parameters{
+					BlockParameters: tosca.BlockParameters{
+						Revision: test.revision,
+					},
+				},
+				stack:  NewStack(),
+				memory: NewMemory(),
+				gas:    3,
+			}
+
+			ctxt.stack.push(uint256.NewInt(0)) // salt
+			ctxt.stack.push(&test.size)
+			ctxt.stack.push(&test.offset)
+			ctxt.stack.push(uint256.NewInt(0)) // value
+
+			err := genericCreate(&ctxt, test.kind)
+			if err != test.expectedError {
+				t.Errorf("unexpected err. wanted %v, got %v", test.expectedError, err)
+			}
+		})
+	}
+}
+
+func TestGenericCreate_ResultIsWrittenToStack(t *testing.T) {
+	CreatedAddress := tosca.Address([20]byte{19: 0x1})
+	for _, success := range []bool{true, false} {
+		runContext := tosca.NewMockRunContext(gomock.NewController(t))
+		runContext.EXPECT().Call(gomock.Any(), gomock.Any()).Return(tosca.CallResult{Success: success, CreatedAddress: CreatedAddress}, nil)
+		ctxt := getEmptyContext()
+		ctxt.context = runContext
+		ctxt.stack.stackPointer = 3
```

Comment:
Careful: there is no guarantee that the stack will only contain zero-elements. This lead to flaky tests in the past. I think here you depend on all 3 elements to be zero. Thus, you should push 3 zeros on the stack instead.